consensus/src/detail/pow/work.rs

```diff
@@ -84,24 +94,37 @@ impl PoW {
             };
         }
 
-        let retarget_block_time = get_starting_block_time(adjustment_interval, prev_block_index);
-        self.next_work_required(retarget_block_time, prev_block_index)
+        let retarget_block_time =
+            get_starting_block_time(adjustment_interval, prev_block_index, db_accessor)?;
+        self.next_work_required(retarget_block_time, prev_block_index, db_accessor)
     }
 
     /// retargeting proof of work
     fn next_work_required(
         &self,
         retarget_block_time: u32,
         prev_block_index: &BlockIndex,
+        db_accessor: &dyn BlockIndexDBAccessor,
     ) -> Result<Compact, Error> {
         // limit adjustment step
         let actual_timespan_of_last_interval =
             self.actual_timespan(prev_block_index.get_block_time(), retarget_block_time);
 
+        let prev_block_bits = {
+            if let ConsensusData::PoW(pow_data) = db_accessor
+                .get_block(prev_block_index.get_block_id().to_owned())?
```

Comment:
Reading the whole block is crazy. This information should be in the block index.